lib/client.js

```diff
@@ -335,29 +333,26 @@ Object.assign(Client.prototype, {
   },
 
   // Acquire a connection from the pool.
-  acquireConnection() {
+  async acquireConnection() {
     if (!this.pool) {
-      return Bluebird.reject(new Error('Unable to acquire a connection'));
-    }
-    try {
-      return Bluebird.try(() => this.pool.acquire().promise)
-        .then((connection) => {
-          debug('acquired connection from pool: %s', connection.__knexUid);
-          return connection;
-        })
-        .catch((error) => {
-          let convertedError = error;
-          if (error instanceof TimeoutError) {
-            convertedError = new KnexTimeoutError(
-              'Knex: Timeout acquiring a connection. The pool is probably full. ' +
-                'Are you missing a .transacting(trx) call?'
-            );
-          }
-          throw convertedError;
-        });
-    } catch (e) {
-      return Bluebird.reject(e);
+      return Promise.reject(new Error('Unable to acquire a connection'));
```

Comment:
No need for an explicit `Promise.reject` here. Since this is an `async` function, all you have to do is throw the error and the caller will see the promise this function returns become rejected.

Reply:
Done